x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.test.ts

```diff
@@ -0,0 +1,1175 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { resetContext } from 'kea';
+
+import { CredentialsLogic } from './credentials_logic';
+import { ADMIN, PRIVATE } from './constants';
+
+jest.mock('../../../shared/http', () => ({
+  HttpLogic: { values: { http: { get: jest.fn(), delete: jest.fn() } } },
+}));
+import { HttpLogic } from '../../../shared/http';
+jest.mock('../../../shared/flash_messages', () => ({
+  flashAPIErrors: jest.fn(),
+}));
+import { flashAPIErrors } from '../../../shared/flash_messages';
+import { IMeta } from '../../../../../common/types';
```

Comment:
New listener tests look great!! This is my last comment for today, and it's just my 2c, not a change request or even remotely a blocker.

I'm probably on the more lazy end of the spectrum here, but IMO, we can get away with using `any` in tests rather than strictly asserting types just to satisfy type checks. So in this case I'd personally feel 100% ok losing this import and doing something like `const meta: any = {` instead.

That being said, I know other devs feel differently, so feel free to do whatever is more comfortable for you (whether it makes writing tests less of a pain/friction point, or whether it gives you more confidence in typing) 😄

Reply:
That's a great point. I'll take a close look tomorrow. As always, thank you for the review.

Reply:
https://github.com/elastic/kibana/pull/77626/commits/1cd681220540fa307f10f241e69b592c25fefd2c